rig-milvus/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+
+            return Err(VectorStoreError::DatastoreError(res.to_string().into()));
+        }
+
+        Ok(())
+    }
+}
+
+impl<M: EmbeddingModel + std::marker::Sync + Send> VectorStoreIndex for MilvusVectorStore<M> {
+    /// Search for the top `n` nearest neighbors to the given query within the Milvus vector store.
+    /// Returns a vector of tuples containing the score, ID, and payload of the nearest neighbors.
+    async fn top_n<T: for<'a> Deserialize<'a> + Send>(
+        &self,
+        query: &str,
+        n: usize,
+    ) -> Result<Vec<(f64, String, T)>, VectorStoreError> {
+        let embedding = self.model.embed_text(query).await?;
+        let url = format!(
+            "{base_url}/v2/vectordb/entities/search",
+            base_url = self.base_url
+        );
+
+        let body = self.create_search_request(embedding.vec, n);
+
+        let mut client = self.client.post(url);
+        if let Some(ref token) = self.token {
+            client = client.header("Authentication", format!("Bearer {token}"));
+        }
+
+        let body = serde_json::to_string(&body).unwrap();
+
+        let res = client
+            .body(body)
+            .send()
+            .await
+            .map_err(|x| VectorStoreError::DatastoreError(x.to_string().into()))?;
+
+        if res.status() != StatusCode::OK {
+            let res = res.text().await.unwrap();
+
+            return Err(VectorStoreError::DatastoreError(res.to_string().into()));
+        }
+
+        let json: SearchResult<T> = res
+            .json()
+            .await
+            .map_err(|x| VectorStoreError::DatastoreError(x.to_string().into()))?;
+
+        let res = json
+            .data
+            .into_iter()
+            .map(|x| (x.distance, x.id.to_string(), x.document))
+            .collect();
+
+        Ok(res)
+    }
+
+    /// Search for the top `n` nearest neighbors to the given query within the Milvus vector store.
+    /// Returns a vector of tuples containing the score and ID of the nearest neighbors.
+    async fn top_n_ids(
+        &self,
+        query: &str,
+        n: usize,
+    ) -> Result<Vec<(f64, String)>, VectorStoreError> {
+        let embedding = self.model.embed_text(query).await?;
+        let url = format!(
+            "{base_url}/v2/vectordb/entities/search",
+            base_url = self.base_url
+        );
+
+        let body = self.create_search_request_id_only(embedding.vec, n);
+
+        let mut client = self.client.post(url);
+        if let Some(ref token) = self.token {
+            client = client.header("Authentication", format!("Bearer {token}"));
+        }
+
+        let body = serde_json::to_string(&body).unwrap();
+
+        let res = client
+            .body(body)
+            .send()
+            .await
+            .map_err(|x| VectorStoreError::DatastoreError(x.to_string().into()))?;
+
+        if res.status() != StatusCode::OK {
+            let res = res.text().await.unwrap();
```

Comment:
Is this `unwrap` safe? I think either use a `.expect` OR probably bubble the error up. I guess it's only when the response is errored but i suppose a default error message could be "failed getting error message".

*tbh, we could think about enabling a `.unwrap` lint in clippy.*